File: API_JeeWork2021/Models/Wework/TemplateModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace JeeWork_Core2021.Models
{
    public class TemplateModel
    {
        public long id_row { get; set; }
        public string title { get; set; }
        public string description { get; set; }
        public string color { get; set; }
        public bool isdefault { get; set; } = true;
        public long customerid { get; set; }
        public List<TemplateStatusModel> Status { get; set; }
    }
    public class UpdateQuickModel
    {
        public long id_row { get; set; }
        public string columname { get; set; } // color, title, statusname
        public string values { get; set; }
        public bool istemplate { get; set; } = true;
        public long customerid { get; set; }
        public long id_template { get; set; } // áp dụng khi thêm status mới cho template
    }
    public class TemplateCenterModel
    {
        public long id_row { get; set; } //
        public string title { get; set; } //
        public long templateid { get; set; } //
        public long customerid { get; set; } //
        public long types { get; set; } // 1 - space, 2 - folder, 3 - list (Project)
        public long ObjectTypesID { get; set; } // Truyền id department hoặc id list
        public long ParentID { get; set; } //2 - folder (Truyền ID Department chọn), 3 - list (Project) Truyền id_Folder chọn
        public long levels { get; set; } // 1 - Beginner, 2 - Intermediate, 3 - Advanced
        public long viewid { get; set; } // 
        public long group_statusid { get; set; } //
        public long template_typeid { get; set; } //
        public long img_temp { get; set; } //
        public long field_id { get; set; } //
        public bool is_customitems { get; set; } = false;
        public bool is_projectdates { get; set; } = false;
        public bool is_task { get; set; } = false;
        public bool is_views { get; set; } = false;
        public List<ListFieldModel> customitems { get; set; }
        public List<ProjectDatesModel> projectdates { get; set; }
    }
    public class ListFieldModel
    {
        public long id_field { get; set; }
        public string fieldname { get; set; }
        public string title { get; set; }
        public bool isvisible { get; set; }
        public string note { get; set; }
        public string type { get; set; }
        public long position { get; set; }
        public bool isnewfield { get; set; }
        public bool isdefault { get; set; }
        public long typeid { get; set; }
    }
    public class ProjectDatesModel
    {
        public DateTime start_date { get; set; }
        public DateTime end_date { get; set; }
    }
}

File: BackEnd/BE_jeework/src/app/pages/WeWork/template-center/template-model/template.model.ts
import { BaseModel } from './../../../../_metronic/jeework_old/core/_base/crud/models/_base.model_new';

export class TemplateModel extends BaseModel {
	id_row: number;
	title: string;
	description: string;
	color: string;
	isdefault: boolean;
	customerid: number;
	Status: Array<TemplateStatusModel> = [];
	clear() {
		this.id_row = 0;
		this.title = "";
		this.description = "";
		this.color = "";
		this.isdefault = true;
		this.customerid = 0;
		this.Status = [];
	}
}

export class UpdateQuickModel {
	id_row: number;
	columname: string;
	values: string;
	istemplate: boolean;
	customerid: number;
	id_template: number;
	clear() {
		this.id_row = 0;
		this.columname = "";
		this.values = "";
		this.istemplate = true;
		this.customerid = 0;
		this.id_template = 0;
	}
}

export class TemplateCenterModel extends BaseModel {
	id_row: number;
	title: string;
	templateid: number;
	customerid: number;
	ObjectTypesID: number;
	ParentID: number;
	types: number;
	levels: number;
	viewid: number;
	group_statusid: number;
	template_typeid: number;
	img_temp: string;
	field_id: string;
	is_customitems: boolean;
	is_task: boolean;
	is_views: boolean;
	is_projectdates: boolean;
	list_field_name: Array<ListFieldModel> = [];
	projectdates: ProjectDatesModel;
	clear() {
		this.id_row = 0;
		this.title = "";
		this.templateid = 0;
		this.customerid = 0;
		this.ObjectTypesID = 0;
		this.ParentID = 0;
		this.types = 0;
		this.levels = 0;
		this.viewid = 0;
		this.group_statusid = 0;
		this.template_typeid = 0;
		this.img_temp = "";
		this.field_id = "";
		this.is_customitems = false;
		this.is_task = false;
		this.is_views = false;
		this.is_projectdates = false;
		this.list_field_name = [];
	}
}

export class ListFieldModel extends BaseModel {
	id_field: number;
	fieldname: string;
	title: string;
	isvisible: boolean;
	note: string;
	type: string;
	position: number;
	isnewfield: boolean;
	isdefault: boolean;
	typeid: number;
	clear() {
		this.id_field = 0;
		this.fieldname = '';
		this.title = '';
		this.isvisible = false;
		this.note = '';
		this.type = '';
		this.position = 0;
		this.isnewfield = false;
		this.isdefault = false;
		this.typeid = 0;
	}
}

export class ProjectDatesModel {
	start_date: string;
	end_date: string;
}
export class TemplateStatusModel extends BaseModel {
	id_row: number;
	title: string;
	is_quahan: boolean;
	urgent: boolean;
	status: boolean;
	clear() {
		this.id_row = 0;
		this.title = '';
		this.is_quahan = false;
		this.urgent = false;
		this.status = false;
	}
}
